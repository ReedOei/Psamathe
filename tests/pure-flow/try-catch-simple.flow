transformer idBool(a : ! bool) -> res : ! bool {
    a --> res
}

transformer notBool(a : ! bool) -> res : ! bool {
    try {
        a --[ ! such that idBool() ]-> var y : ! bool
        false --> res
    } catch {
        true --> res
    }
}

;

false --> var b : ! bool
b --[ ! such that notBool() ]-> var c : ! bool

