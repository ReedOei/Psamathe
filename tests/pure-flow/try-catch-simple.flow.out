<generatedTop>
  <common>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
    <freshCounter>
      7
    </freshCounter>
  </common>
  <k>
    .
  </k>
  <functions>
    idBool |-> transformer ( res : ! bool , a : ! bool , .VarDefs , a --> res  .Stmts )
    notBool |-> transformer ( res : ! bool , a : ! bool , .VarDefs , try { a [ ! such that idBool ( .Vars ) ] --> var y : bool  false --> res  .Stmts } catch { true --> res  .Stmts }  .Stmts )
  </functions>
  <lookup>
    b |-> loc ( 0 )
    c |-> loc ( 2 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( bool , .List )
    loc ( 1 ) |-> resource ( bool , .List )
    loc ( 2 ) |-> resource ( bool , ListItem ( false ) )
    loc ( 3 ) |-> resource ( bool , ListItem ( true ) )
    loc ( 6 ) |-> resource ( bool , .List )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
